src/lib/io/sol-iio.c

```diff
@@ -43,6 +43,7 @@
 #endif
 
 struct sol_iio_device {
+    double *mount_matrix;
```

Comment:
I don't get it. Even as a pointer you will always alloc an array of size 9.

Quoting your code:
```C
device->mount_matrix = calloc(1, sizeof(double) * MOUNT_MATRIX_LEN);
```